hw7/hw7_t03.py

```diff
@@ -0,0 +1,70 @@
+"""
+Given a Tic-Tac-Toe 3x3 board (can be unfinished).
+
+Write a function that checks if the are some winners.
+If there is "x" winner, function should return "x wins!"
+If there is "o" winner, function should return "o wins!"
+If there is a draw, function should return "draw!"
+If board is unfinished, function should return "unfinished!"
+Example:
+    [[-, -, o],
+     [-, x, o],
+     [x, o, x]]
+    Return value should be "unfinished"
+    [[-, -, o],
+     [-, o, o],
+     [x, x, x]]
+     Return value should be "x wins!"
+"""
+from typing import List, Optional
+
+
+def check_row_win(board: List[List]) -> Optional[str]:
+    finished = True
+    for row in range(3):
+        if board[row].count("-") > 0:
+            finished = False
+        elif board[row].count("x") == 3:
+            return "x wins!"
+        elif board[row].count("o") == 3:
+            return "o wins!"
+    if not finished:  # noqa: R503
+        return "unfinished"
+
+
+def check_column_win(board: List[List]) -> Optional[str]:
+    for col in range(3):  # noqa: R503
+        if [board[row][col] for row in range(3)].count("x") == 3:
+            return "x wins!"
+        elif [board[row][col] for row in range(3)].count("o") == 3:
+            return "o wins!"
+
+
+def check_diagonal_win(board: List[List]) -> Optional[str]:
+    if [board[index][index] for index in range(3)].count("x") == 3:
+        return "x wins!"
+    elif [board[index][index] for index in range(3)].count("o") == 3:
+        return "o wins!"
+    elif [board[index][2 - index] for index in range(3)].count("x") == 3:
+        return "x wins!"
+    elif [board[index][2 - index] for index in range(3)].count("o") == 3:  # noqa: R503
+        return "o wins!"
+
+
+def tic_tac_toe_checker(board: List[List]) -> str:
+    finished = True
+    row_check_result = check_row_win(board)
```

Comment:
I'd say it is a little bit confusing that `check_diagonal_win` and `check_column_win` have a single responsibility and `check_row_win` has 2 of them.

Reply:
I made some refactoring. There will be 1 more run-through on the board, but the code is much cleaner now.